pandas/core/arrays/string_.py

```diff
@@ -182,6 +182,16 @@ def _from_sequence(cls, scalars, dtype=None, copy=False):
     def _from_sequence_of_strings(cls, strings, dtype=None, copy=False):
         return cls._from_sequence(strings, dtype=dtype, copy=copy)
 
+    def __arrow_array__(self, type=None):
```

Comment:
shouldn't we make a new base EA mixin class for this? e.g. this is pretty generic code

Reply:
and have String/Integer array use this? (maybe actually just put in the EA base class itself?)

Reply:
For now, this one is different than the one in IntegerArray, so it cannot directly be shared with that. I thought about putting this in the PandasArray class, but the StringArray is the only subclass of it that is actually used. So therefore I kept it here. But open to other options.

Reply:
> maybe actually just put in the EA base class itself?

That's not possible, because it depends on the internals (eg StringArray uses `_ndarray` for the data, `IntegerArray` uses `_data` and `_mask`). The only thing that could be put in the base EA class is the signature with a NotImplementedError, but I need to check how that interacts with pyarrow (as the code there does a hasattr check, so if the method is present, it expects it to work)